Zend/zend_API.h

```diff
@@ -843,6 +843,12 @@ static zend_always_inline void zend_call_known_fcc(
 	const zend_fcall_info_cache *fcc, zval *retval_ptr, uint32_t param_count, zval *params, HashTable *named_params)
 {
 	zend_function *func = fcc->function_handler;
+	if (UNEXPECTED(!func)) {
```

Comment:
This should be handled in curl, and not call `zend_call_known_fcc` instead. I don't really know curl, but it looks like you might want to unset the `PHP_CURL_USER` flag when it's set to `null`.